pkg/proc/arch.go

```diff
@@ -302,5 +304,10 @@ func (a *AMD64) GoroutineToDwarfRegisters(g *G) op.DwarfRegisters {
 	dregs[amd64DwarfIPRegNum] = op.DwarfRegisterFromUint64(g.PC)
 	dregs[amd64DwarfSPRegNum] = op.DwarfRegisterFromUint64(g.SP)
 	dregs[amd64DwarfBPRegNum] = op.DwarfRegisterFromUint64(g.BP)
-	return op.DwarfRegisters{StaticBase: g.variable.bi.staticBase, Regs: dregs, ByteOrder: binary.LittleEndian, PCRegNum: amd64DwarfIPRegNum, SPRegNum: amd64DwarfSPRegNum, BPRegNum: amd64DwarfBPRegNum}
+
+	bi := g.variable.bi
```

Comment:
I agree but I think this is alright for now. When (and if) we ever run into a situation where a linear search through loaded shared objects becomes a problem we can tackle it.

Reply:
Added the helper function. 

>This should overall be more efficient than simply using the PC to find the compilation unit and thus the image because we can binary search function entries and we're doing a linear search of compilation unit PC ranges, correct?

I think so. Ideally we'd just do a binary search on images but I haven't figured out how to find the end address.